### PR TITLE
Use the new typed fetchers for getExhibitions

### PR DESCRIPTION
**Pull request by alexwlchan**
For #7363 / #7426

This replaces the getExhibitions function, which allows us to start cleaning up the UiExhibition type also. I'm having weird type checker errors locally, so pushing into CI to see if that fixes it.

### CODE REVIEW

File: content/webapp/pages/whats-on.tsx [hunk 1]
```diff
@@ -21,7 +23,7 @@ import {
 } from '@weco/common/services/prismic/opening-times';
 import {
   cafePromo,
-  shopPromo,
+  // shopPromo,
```
Comment (alexwlchan) [inline content/webapp/pages/whats-on.tsx:26]: I have no idea why TypeScript is suddenly angry about this – it's used in some commented-out code further down, which is why I haven't removed it entirely.

File: content/webapp/components/ExhibitionsAndEvents/ExhibitionsAndEvents.tsx [hunk 2]
```diff
@@ -1,16 +1,17 @@
-import { UiExhibition } from '@weco/common/model/exhibitions';
+import { Exhibition } from '@weco/common/model/exhibitions';
 import { UiEvent } from '@weco/common/model/events';
 import { Link } from '@weco/common/model/link';
 import CardGrid from '../CardGrid/CardGrid';
+import { FunctionComponent } from 'react';
 
 type Props = {
-  exhibitions: UiExhibition[];
+  exhibitions: Exhibition[];
```
Comment (alexwlchan) [inline content/webapp/components/ExhibitionsAndEvents/ExhibitionsAndEvents.tsx:8]: These changes are because the new `transform()` method is returning an `Exhibition` instead of a `UiExhibition`. I strongly suspect the latter is redundant and could be safely deleted, but that'll be a separate patch.

**Review by jamieparkinson — APPROVED** [review]
